src/porepy/viz/data_saving_model_mixin.py

```diff
@@ -10,52 +10,24 @@
 from __future__ import annotations
 
 from pathlib import Path
-from typing import Any, Callable, Optional, Union
+from typing import Optional, Union
 
 import numpy as np
 
 import porepy as pp
+from porepy.models.protocol import PorePyModel
 from porepy.viz.exporter import DataInput
 
 
-class DataSavingMixin:
+class DataSavingMixin(PorePyModel):
     """Class for saving data from a simulation model.
 
     Contract with other classes:
         The model should/may call save_data_time_step() at the end of each time step.
 
     """
 
-    equation_system: pp.EquationSystem
-    """Equation system manager."""
-    params: dict[str, Any]
-    """Dictionary of parameters. May contain data saving parameters."""
-    time_manager: pp.TimeManager
-    """Time manager for the simulation."""
-    mdg: pp.MixedDimensionalGrid
-    """Mixed dimensional grid for the simulation."""
-    restart_options: dict
-    """Dictionary of parameters for restarting from pvd."""
-    units: pp.Units
-    """Units for the simulation."""
-    fluid: pp.FluidConstants
-    """Fluid constants for the simulation."""
-    nd: int
-    """Number of spatial dimensions for the simulation."""
-    nonlinear_solver_statistics: pp.SolverStatistics
-    """Non-linear solver statistics for nonlinear solver."""
-
     def save_data_time_step(self) -> None:
-        """Export the model state at a given time step, and log time.
-        The options for exporting times are:
-            * None: All time steps are exported
```

Comment:
I don't think we want to delete the docstrings from these methods, even though they are also present in the protocol. Do you agree @Yuriyzabegaev?

Reply:
This applies to several methods in this file.

Reply:
Yes, the current convention is to duplicate docstrings. This is something we can mention on the next user meeting.

Reply:
This change was not intended, may have been caused by the messy rebase. I will check for other missing docstrings.